pkgs/development/interpreters/jruby/default.nix

```diff
@@ -1,67 +1,115 @@
-{ lib, stdenv, callPackage, fetchurl, makeWrapper, jre }:
+{ lib, stdenv, callPackage, fetchFromGitHub, makeWrapper, jre, maven, ant, jrubyVersion ? "9.2.16.0" }:
 
 let
-# The version number here is whatever is reported by the RUBY_VERSION string
-rubyVersion = callPackage ../ruby/ruby-version.nix {} "2" "5" "7" "";
-jruby = stdenv.mkDerivation rec {
-  pname = "jruby";
+  # The version number here is whatever is reported by the RUBY_VERSION string
+  rubyVersion = callPackage ../ruby/ruby-version.nix { } "2" "5" "7" "";
+  jruby-src = fetchFromGitHub {
+    owner = "jruby";
+    repo = "jruby";
+    rev = jrubyVersion;
+    sha256 = "14zs1xr9rb7ydna7kfr2xvsvgwpm8lr5ngmb8dqgdv1hq687h6wz";
+  };
+  # Nixpkgs works by disallowing Internet access in order to enforce
+  # reproducibility. One way around that however is if your build is already
+  # binary reproducible, which you can toggle by specifying outputHash.
+  # We create a maven repository (~/.m2) for use in the build further down.
+  maven-repository = stdenv.mkDerivation {
+    name = "jruby-maven-repository";
+    buildInputs = [ jre maven ];
 
-  version = "9.2.16.0";
+    src = jruby-src;
+    buildPhase = ''
+      mkdir $out
+      mvn -Pdist -Dmaven.repo.local=$out -Dmaven.wagon.rto=5000
+    '';
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      find $out -type f -name '*.lastUpdated' -delete
+      find $out -type f -name resolver-status.properties -delete
+      find $out -type f -name _remote.repositories -delete
 
-  src = fetchurl {
-    url = "https://s3.amazonaws.com/jruby.org/downloads/${version}/jruby-bin-${version}.tar.gz";
-    sha256 = "sha256-WuJ/FJ9z8/6k80NZy7dzwl2dmH5yte3snouTlXmX6zA=";
-  };
+      # This check here is placed because I had trouble with find actually deleting the
+      # desired files. Originally, I had used finds `-or` command to chain the above
+      # but it was failing to perform the desired action. This is a little sanity check.
+      lines=$(find $out -type f \
+        -name '*.lastUpdated' -or \
+        -name resolver-status.properties -or \
+        -name _remote.repositories | wc -l)
+      if [ $lines -ne 0 ]; then
+        echo "Failed to delete files that cause output hash to change"
+        exit 1
+      fi
```

Comment:
This was super annoying.
Originally the `find` command looked something like this:

```find $out -type f \
        -name '*.lastUpdated' -or \
        -name resolver-status.properties -or \
        -name _remote.repositories
        -delete```

That was in fact not deleting the files though... it was a bit of a head scratcher.